core/src/validator.rs

```diff
@@ -827,6 +829,8 @@ impl Validator {
     }
 
     pub fn join(self) {
+        drop(self.cluster_info);
```

Comment:
Why do we need `drop` here?

Reply:
I'm just being defensive.  Some of the validator stages are sensitive to when drops occur.  For example of you hold a blockstore in `Validator` and try to `Validator::join()`, retransmit stage will never terminate.  So by dropping the new cluster_info reference here first, I'm 100% sure the stage shutdown path will remain unchanged.  Not worth it?

Reply:
I see, thanks for the explanation.

Reply:
> maybe it implies that there are some circular references

Yeah for sure.  In the blockstore case, it holds a channel that stages assume get dropped.  Without that drop their thread loops never terminate.